pennylane/templates/constructors/broadcast_double.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2018-2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``broadcast`` template constructor.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from collections import Iterable
+
+from pennylane.templates.decorator import template
+from pennylane.templates.utils import _check_wires, _check_type, _get_shape
+
+
+@template
+def broadcast_double(block, wires, parameters=None, even=True, kwargs={}):
+    r"""Applies a two-qubit unitary to pairs of wires.
```

Comment:
```suggestion
    r"""Applies the same two-qubit unitary ``block`` to pairs of wires.
```

Reply:
It might be worth elaborating slightly more on the logic (however simple it is, when grasped) right at the beginning. Reading for the very first time, it felt slightly as getting into the explanation of the wire-business pretty quickly. :)

This could also be aided by further extending the first sentence (or potentially elaborating on parts of it). The first sentence could also go like "Broadcasts the same two-qubit unitary by applying the same  ``block`` to pairs of wires.". Though these are just ideas for now, not too sure which would be the best to have.

Reply:
I tried to implement this advice in the now single ``broadcast`` function.